### DISCUSSION TITLE
Just the class vs. Morea Framework

### DISCUSSION THREAD

**Discussion by philipmjohnson** [kevinlin1/just-the-class#15] [Show and tell]
I just came across Just The Class and thought, well, this is an interesting example of convergent evolution!

I built a very similar framework 5 years ago called Morea:

https://morea-framework.github.io/

Morea seems to be a little bit more "module" oriented than JTC. But otherwise the overlap in philosophy is quite striking.

**Comment by kevinlin1** [comment]
Neat! The "module" organization is pretty common among learning management systems.

**Reply by philipmjohnson** [reply]
Well, there's a little more to modules than meets the eye. Here's what Blackboard (for example) has as modules:

https://help.blackboard.com/Learn/Instructor/Ultra/Course_Content/Create_Content/Create_Containers_for_Content/Create_Learning_Modules

Basically, "modules" in Blackboard are simply "folders" for unstructured content. 

There are two significant design aspects of modules in Morea that are not present in LMS systems like Blackboard:

1. Morea modules are "typed".   In Morea, a "module" is a collection of Outcomes, Readings, Experiences, and Assessments. This is significant because it enables a simple form of quality assurance on your pedagogy. For example, are all of your Outcomes covered by at least one Assessment?  Do you have an appropriate mix of passive (Readings) and active (Experiences) homework assignments?   On the other hand, Blackboard modules are simply "containers" without any "semantics". 

2. Morea modules are designed to facilitate reuse, sharing, and evolution.  Morea modules, conceptually, are like npm packages: self-contained units of pedagogy that can be easily shared across different sites and instructors. They are like source code.   Blackboard modules remind me more of formulas in an excel spreadsheet: yes, you can "share" them, but it's painful and the design actively works against you if you try to do it.

What I like about JTC is that you've gone the same direction as Morea: GitHub for back-end storage and hosting, Jekyll for compilation, markdown as the source language, and (presumably) the "local" LMS for other things like the grade book.  This makes JTC more like "source code" rather than a "spreadsheet formula", and it's hard to go back to the latter once you've experienced the former.
